parlai/core/utils.py

```diff
@@ -181,6 +181,24 @@ def load_opt_file(optfile):
     return Opt(opt)
 
 
+class Message(dict):
+    """
+    Class for observations and actions.
+
+    Functions like a dict, but warns when writing to fields that already exist.
+    """
+    def __setitem__(self, key, val):
+        if key in self:
+            warn_once(
+                'Message already contains key `{}`. Please make sure '
+                'this change is intentional.'.format(key)
```

Comment:
I think this warning is a little too weak to inform users of what's going on when they write to these fields. As an end user I might read this and say "yeah, I intended to change the value, that's why I did it" and disregard. Something like a second sentence explaining the risks of mutating a `Message` object before the 'please make sure' might be valuable.

Reply:
Maybe something along the lines of `Agents may edit fields in unexpected ways`?

Reply:
I'm thinking more noting possible consequences of making edits to the Message object. Even one concrete example can provide enough context as to why editing this dict can be a bad thing if you don't know what you're doing. `Unexpected ways` doesn't really lead me to understanding when it makes sense to make intentional changes to this and when it could ruin my experiments or cause issues. Of course, its possible that we'll want to go more in-depth into the implementation details somewhere else.